packages/fetchai/skills/confirmation_aw1/strategy.py

```diff
@@ -152,18 +162,50 @@ def valid_registration(
         :param sender: the sender
         :return: tuple of success, error code and error message
         """
-        if not all([key in registration_info for key in REQUIRED_KEYS]):
+        if not all([key in registration_info for key in DEVELOPER_ONLY_REQUIRED_KEYS]):
```

Comment:
the required keys should be a default arg to the function as you'll need to override it below.

Reply:
The logic is slightly different here from what you have in mind. 

`_valid_registration_developer_only` contains all checks needed for the developer only mode. Then in `valid_registration` we call `_valid_registration_developer_only`. If we are in developer_only mode, we've done all checks so just return the results. If not, those checks are a prerequisite, so we continue with the other checks and return the overall result.